pkg/api/customization/clustertemplate/cluster_template.go

```diff
@@ -70,10 +87,101 @@ func (w Wrapper) LinkHandler(apiContext *types.APIContext, next types.RequestHan
 }
 
 func (w Wrapper) ClusterTemplateRevisionsActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
-	if actionName != "listquestions" {
-		return httperror.NewAPIError(httperror.NotFound, "not found")
+
+	canUpdateClusterTemplateRevision := func() bool {
+		revision := map[string]interface{}{
+			"id": apiContext.ID,
+		}
+
+		return apiContext.AccessControl.CanDo(v3.ClusterTemplateRevisionGroupVersionKind.Group, v3.ClusterTemplateRevisionResource.Name, "update", apiContext, revision, apiContext.Schema) == nil
+	}
+
+	switch actionName {
+	case "disable":
+		if !canUpdateClusterTemplateRevision() {
+			return httperror.NewAPIError(httperror.PermissionDenied, "can not access clusterTemplateRevision")
+		}
+		return w.disableRevision(actionName, action, apiContext)
+	case "enable":
+		if !canUpdateClusterTemplateRevision() {
+			return httperror.NewAPIError(httperror.PermissionDenied, "can not access clusterTemplateRevision")
+		}
+		return w.enableRevision(actionName, action, apiContext)
+	case "listquestions":
+		return w.listRevisionQuestions(actionName, action, apiContext)
+
 	}
+	return httperror.NewAPIError(httperror.NotFound, "not found")
+}
+
+func (w Wrapper) disableRevision(actionName string, action *types.Action, apiContext *types.APIContext) error {
```

Comment:
`enableRevision` and `disableRevision` could be refactored into just one function passing the bool enabled value